flagd/pkg/service/flag-sync/sync_service.go

```diff
@@ -39,14 +43,38 @@ type Service struct {
 	startupTracker syncTracker
 }
 
+func loadTLSCredentials(certPath string, keyPath string) (credentials.TransportCredentials, error) {
+	// Load server's certificate and private key
+	serverCert, err := tls.LoadX509KeyPair(certPath, keyPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load key pair from given certificate paths: '%s' and '%s'", certPath, keyPath)
+	}
+
+	// Create the credentials and return it
+	config := &tls.Config{
```

Comment:
To get rid of this error, you want to make the minimum TLS version 1.2. Just add:

`MinVersion: tls.VersionTLS12` to the options.

I don't think anyone will require anything less (and we can always add an option for it if so).